sklearn/decomposition/_dict_learning.py

```diff
@@ -735,6 +735,42 @@ def _check_warn_deprecated(param, name, default, additional_message=None):
         return default
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
```

Comment:
It seems that the documentation is stating `ndarray`. I assume that we could use `array-like` but we need to make sure that `X` is validated when `max_iter is not None`, corresponding to the case when we don't call the class.

Reply:
I'm not sure how to solve this. Changing `"X": [np.ndarray]` causes an error in the test function `TypeError: ndarray() missing required argument 'shape' (pos 1)` but I don't understand why.
How can we validate `X` when `max_iter is not None`? What exactly should be validated?